routes/chat.js

```diff
@@ -6,11 +6,13 @@ const chatController = require('../controllers/chat/ChatController');
 const auth = require('../middlewares/auth');
 const isTargetUserAllowingAnonDMMiddleware = require('../middlewares/chat/isTargetUserAllowingAnonDMMiddleware');
 const SuccessMiddleware = require('../middlewares/success');
+const sendSignedMessage = require('../controllers/chat/sendSignedMessage');
 
 router.get('/create-channel', chatController.createChannel);
 router.post('/add-moderator', chatController.addChannelModerator);
 router.post('/add-members-channel', auth.isAuth, chatController.addMembers);
-router.post('/anonymous', auth.isAuth, chatController.sendAnonymous);
+router.post('/anonymous', auth.isAuthAnonim, chatController.sendAnonymous);
+router.post('/send-signed-message', auth.isAuthV2, sendSignedMessage);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are good for separating concerns and improving the modularity of the code. The `sendSignedMessage` function has been moved to its own controller, which is a good practice for maintaining separation of concerns. Also, the use of different authentication middlewares (`auth.isAuthAnonim` and `auth.isAuthV2`) for different routes helps in enforcing appropriate access controls based on user roles or types.

However, I noticed that there's no middleware to check if the user is authorized to send a signed message. It would be a good idea to add an authorization middleware to ensure that only users with the necessary permissions can send signed messages.

```diff
- router.post('/send-signed-message', auth.isAuthV2, sendSignedMessage);
+ router.post('/send-signed-message', auth.isAuthV2, auth.canSendSignedMessage, sendSignedMessage);
```
In the above diff, `auth.canSendSignedMessage` is a hypothetical middleware function that checks if the authenticated user has the necessary permissions to send a signed message. You would need to implement this function according to your application's specific authorization rules.


<!-- This is an auto-generated comment by OSS CodeRabbit -->